CMakeLists.txt

```diff
@@ -438,7 +448,7 @@ endif()
 
 set(USE_X11 0)
 
-if(UNIX AND NOT APPLE AND NOT ANDROID AND NOT ENABLE_HEADLESS)
+if(UNIX AND NOT APPLE AND NOT ANDROID AND NOT HAIKU AND NOT ENABLE_HEADLESS)
```

Comment:
I don't think it's my place to do that (and I was trying to keep this PR to a minimum; I already moved some OS-independent changes into another PR which was already merged)